GDJS/Runtime/howler-sound-manager/howler-sound-manager.js

```diff
@@ -14,87 +15,263 @@
  * @memberof gdjs
  * @class HowlerSound
  */
-gdjs.HowlerSound = function (o) {
-  Howl.call(this, o);
-  this._paused = false;
-  this._stopped = true;
-  this._canBeDestroyed = false;
-  this._rate = o.rate || 1;
+gdjs.HowlerSound = class HowlerSound {
+  _id = null;
+  _howl = null;
+  _paused = false;
+  _stopped = true;
+  _canBeDestroyed = false;
+
+  constructor(howl) {
+    this._howl = howl;
+
+    /*
+     * Track events to be sure the status is
+     * sync'ed with the sound - though some should be redundant
+     * with `play`/`pause` methods already doing that. Keeping
+     * that to be sure that the status is always correct.
+     */
+
+    this._howl.on('play', function () {
+      this._paused = false;
+      this._stopped = false;
+    });
+
+    this._howl.on('pause', function () {
+      this._paused = true;
+      this._stopped = false;
+    });
+
+    this._howl.on('end', () => {
+      if (!this._howl.loop()) {
+        this._canBeDestroyed = true;
+        this._paused = false;
+        this._stopped = true;
+      }
+    });
+
+    this._howl.on('playerror', (id, error) => {
+      console.error(
+        "Can't play a sound, considering it as stopped. Error is: ",
+        error
+      );
+      this._paused = false;
+      this._stopped = true;
+    });
+
+    // In case the howl didn't finsih loading, start the sound once it finishes loading.
+    this._howl.on('load', function () {
+      this.play();
+    });
```

Comment:
All these events are attached to `howl`, which is shared between all the sounds (or musics) of the same resource, the difference being done using the `id` - but these events are not looking at the id. So I think we risk, when stopping/pausing/playing a sound, to mark all the sounds with the same resource as stopped/paused/playing.
(this is the kind of edge case that I'm a bit afraid of, and why it would be a very good idea to have a test game where you can start multiple sounds or musics, stop them, pause them, etc..).

I think `id` is passed to each callback, in which case you must check if the id is equal to `this._id` - hopefully that is all that is needed.

Reply:
This is more problematic, as at this point we don't have the id yet as it is defined only after begining playback...